pkg/server/proto/server.proto

```diff
@@ -1143,6 +1159,47 @@ message Task {
   opaqueany.Any state = 6;
 }
 
+message UpsertTaskRequest {
+  // The Task to upsert. If the id in the Task message is empty, then this will
+  // be an insert in the db. Otherwise if an id was requested, this will act
+  // as an update. If the ID isn't found, it will be an error.
+  Task task = 1;
+}
+
+message UpsertTaskResponse {
+  // The resulting Task that was Upserted.
+  Task task = 1;
+}
+
+message GetTaskRequest {
+  // Can be oneof Task id or by the Run job id.
+  Ref.Task ref = 1;
+}
+
+message GetTaskResponse {
```

Comment:
I wonder if we want some sort "active_job" here that is an enum like

```
PENDING
STARTING <== start job is running
STARTED <== start job complete, task is not running yet
RUNNING <== task is running
COMPLETED <== task is done
STOPPING <== stop job is running
STOPPED
```

Cause we'd want to use this kind of information for the job system probably. Cause there are various valid state transitions here we'd want to handle like `STARTED => STOPPED` (canceled before task is executed) to clean up resources. Or `PENDING => STOPPED` (easy, metadata update, just cancel the jobs since no resources created). etc.